src/libs/fileDownload/index.android.js

```diff
@@ -41,7 +41,7 @@ function handleDownload(url, fileName) {
 
         // Android files will download to Download directory
         const path = dirs.DownloadDir;
-        const attachmentName = fileName || FileUtils.getAttachmentName(url);
+        const attachmentName = FileUtils.appendTimeToFileName(fileName) || FileUtils.getAttachmentName(url);
```

Comment:
The current logic for determining `attachmentName` doesn't adequately handle fallback scenarios:

```js
const attachmentName = FileUtils.appendTimeToFileName(fileName) || FileUtils.getAttachmentName(url);
```

1. If `fileName` is `undefined`, the application will throw an error due to the attempt to append time to it, leading to a failed download.
2. `FileUtils.appendTimeToFileName` always returns a value, so we won't ever fall back to `FileUtils.getAttachmentName`, even in cases where `fileName` is an empty string.

The previous version of the function handled both of these cases

Reply:
I've stumbled on the problem, when I was testing my PR on iOS: https://github.com/Expensify/App/pull/25556

I've already included a fix in my PR 

It seems `fileName` can be undefined for attachments that are send by Concierge, when they embed images via Drag & Drop